lightning/src/ln/onion_utils.rs

```diff
@@ -413,15 +416,18 @@ pub(super) fn process_onion_failure<T: secp256k1::Signing, L: Deref>(secp_ctx: &
 											20 => chan_update.contents.flags & 2 == 0,
 											_ => false, // unknown error code; take channel_update as valid
 										};
-										network_update = if is_chan_update_invalid {
+										if is_chan_update_invalid {
 											// This probably indicates the node which forwarded
 											// to the node in question corrupted something.
-											Some(NetworkUpdate::ChannelClosed {
+											network_update = Some(NetworkUpdate::ChannelClosed {
 												short_channel_id: route_hop.short_channel_id,
 												is_permanent: true,
-											})
+											});
 										} else {
-											Some(NetworkUpdate::ChannelUpdateMessage {
+											if failing_route_hop.short_channel_id == chan_update.contents.short_channel_id {
```

Comment:
I think a comment here could be useful, to know the reasoning for when to set `short_channel_id`?

Reply:
Done.